tools/walletextension/useraccountmanager/user_account_manager.go

```diff
@@ -34,14 +42,61 @@ func (m *UserAccountManager) AddAndReturnAccountManager(userID string) *accountm
 }
 
 // GetUserAccountManager retrieves the UserAccountManager associated with the given userID.
-// It returns the UserAccountManager and nil error if one exists.
+// it returns the UserAccountManager and nil error if one exists.
+// before returning it checks the database and creates all missing clients for that userID
+// (we are not loading all of them at startup to limit the number of established connections)
 // If a UserAccountManager does not exist for the userID, it returns nil and an error.
 func (m *UserAccountManager) GetUserAccountManager(userID string) (*accountmanager.AccountManager, error) {
 	userAccManager, exists := m.userAccountManager[userID]
-	if exists {
+	if !exists {
+		return nil, fmt.Errorf("UserAccountManager doesn't exist for user: %s", userID)
+	}
+
+	// we have userAccountManager as expected.
+	// now we need to create all clients that don't exist there yet
+	addressesWithClients := userAccManager.GetAllAddressesWithClients()
+
+	// get all addresses for current userID
+	userIDbytes, err := hex.DecodeString(userID)
+	if err != nil {
+		return nil, err
+	}
+
+	// log that we don't have a storage, but still return existing userAccountManager
+	if m.storage == nil {
+		m.logger.Error("storage is nil in UserAccountManager")
 		return userAccManager, nil
 	}
```

Comment:
This should never happen - but if it's used for tests then it's fine we just need a comment

Reply:
added a comment